lib/kadm5/chpass_c.c

```diff
@@ -59,36 +59,49 @@ kadm5_c_chpass_principal(void *server_handle,
        return KADM5_KS_TUPLE_NOSUPP;
 
     ret = _kadm5_connect(server_handle);
-    if(ret)
+    if (ret)
 	return ret;
 
+    krb5_data_zero(&reply);
+
     sp = krb5_storage_from_mem(buf, sizeof(buf));
     if (sp == NULL) {
-	krb5_clear_error_message(context->context);
-	return ENOMEM;
+	ret = ENOMEM;
```

Comment:
`ret = krb5_enomem(context);` would be better -- that's what `krb5_enomem()` exists for.